java/src/main/java/com/nitorcreations/vault/VaultClient.java

```diff
@@ -37,6 +39,11 @@
   private static final String VALUE_OBJECT_NAME_FORMAT = "%s.%s";
   private static final String KEY_OBJECT_NAME_FORMAT = "%s.key";
 
+  @Deprecated
+  public VaultClient(AmazonS3Client s3, AWSKMSClient kms, String bucketName, String vaultKey) {
```

Comment:
The new constructor works also when calling with concrete classes.

Reply:
Yes, but having the deprecated constructor makes it clear that people should change their code